lib/modules/datasource/go/base.ts

```diff
@@ -123,9 +123,8 @@ export class BaseGoDatasource {
         // get server base url from import url
         const parsedUrl = URL.parse(goSourceUrl);
 
-        // split the go module from the URL: host/go/module -> go/module
-        const split = goModule.split('/');
-        const packageName = split[1] + '/' + split[2];
+        // trim leading slash from pathname to obtain the package name
+        const packageName = parsedUrl.pathname.replace(/^\//, '');
```

Comment:
use `trimTrailingSlash` from `util/url` 😉

Reply:
It's a leading slash though, not sure we have any util for this yet

Reply:
I haven't found anything. I could add
```
export function trimLeadingSlash(path: string): string {
  return path.replace(/^\//, '');
}
```
to `util/url`.